flink-streaming-connectors/flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java

```diff
@@ -0,0 +1,143 @@
+	private static final Logger LOG = LoggerFactory.getLogger(RedisSink.class);
+
+	private static final int DEFAULT_TIMEOUT = 2000;
+
+	private String host;
+	private int port;
+	private String channel;
+	private int timeOut;
+	private int soTimeOut;
+	private int database;
+	private String password;
+	private String clientName;
+
+	private SerializationSchema<IN> schema;
+	private transient JedisPoolConfig poolConfig;
+	private transient JedisPool jedisPool;
+
+	/**
+	 * Creates a new RedisSink. For passing custom connection Pool config, please use the constructor
+	 * {@link RedisSink#RedisSink(String, int, String, SerializationSchema, int, int, String, int, String,
+	 * JedisPoolConfig poolConfig)},
+	 * @param host Redis Host name to connect to
+	 * @param port Redis instance Port
+	 * @param channel The channel to which data will be published
+	 * @param schema A {@link SerializationSchema} for turning the java object to bytes.
+	 */
+	public RedisSink(String host, int port,  String channel, SerializationSchema<IN> schema) {
+		this(host, port, channel, schema, DEFAULT_TIMEOUT, DEFAULT_TIMEOUT, null, Protocol.DEFAULT_DATABASE, null, null);
+	}
+
+	/**
+	 * Creates a new RedisSink.
+	 * @param host Redis Host name to connect to
+	 * @param port Redis instance Port
+	 * @param channel The channel to which data will be published
+	 * @param schema A {@link SerializationSchema} for turning the java object to bytes.
+	 * @param timeOut Connection Timeout in millisecond
+	 * @param soTimeOut Socket Timeout in millisecond. Sets Socket::SO_TIMEOUT
+	 * @param password Password for Redis Server
+	 * @param database Select the DB with having the specified zero-based numeric index
+	 * @param clientName Assigns a name to the current connection using CLIENT SETNAME command
+	 * @param poolConfig Custom jedis connection pool configuration
+	 */
+	public RedisSink(String host, int port, String channel, SerializationSchema<IN> schema, int timeOut,
+					int soTimeOut, String password, int database, String clientName, JedisPoolConfig poolConfig) {
+		this.host = host;
+		this.port = port;
+		this.channel = channel;
+		this.schema = schema;
+		this.timeOut = timeOut;
+		this.soTimeOut = soTimeOut;
+		this.password = password;
+		this.database = database;
+		this.clientName = clientName;
+		this.poolConfig = poolConfig;
+	}
+
+	/**
+	 * Called when new data arrives to the sink, and forwards it to Redis channel.
+	 *
+	 * @param value The incoming data
+	 */
+	@Override
+	public void invoke(IN value) throws Exception {
+		try (Jedis jedis = jedisPool.getResource()) {
+			byte[] msg = schema.serialize(value);
+			jedis.publish(channel.getBytes(), msg);
+		} catch (Exception e) {
+			if (LOG.isErrorEnabled()) {
+				LOG.error("Cannot send Redis message {}", channel);
+			}
+		}
+	}
+
+	@Override
+	public void open(Configuration parameters) throws Exception {
+
+		Preconditions.checkNotNull(this.host, "Redis host name should not be Null");
```

Comment:
Why do you check here and not in constructor? Checking in constructor will give error on client side, ie, the error is detected earlier. The check here, would be done in the TaskManager (ie, on the cluster). This is way harder to debug for a user.